lib/puppet_x/puppetlabs/registry/provider_base.rb

```diff
@@ -136,6 +188,23 @@ def name2type(name)
     type2name_map.each_pair {|k,v| name2type[v] = k}
     name2type[name]
   end
+
+  def each_value(key, &block)
+    # TODO: this is fixed in Ruby 2.2, but not 2.1 - is this the best way to check
+    # check via RUBY_VERSION =~ /2.1/
+    # or do we check using responds_to?(Puppet::Util::Windows::Registry.each_value)
+    if registry_helpers.respond_to?(:each_value)
```

Comment:
The question here is whether or not we should bump the `Gemfile` to a newer version of Puppet... 

We will get into a weird situation with failures if:

* Ruby 2.1.5
* This version module code
* Puppet < 4